students/DrewSmith/lessons/lesson01/assignment/tests/test_unit.py

```diff
@@ -0,0 +1,289 @@
+
+    def test_main_menu_option_Q(self):
+        result = main.main_menu(user_prompt="q")
+        self.assertEqual(result, main.exit_program, "Invalid main menu result for option q")
+
+    def test_main_menu_user_select(self):
+        with mock.patch('main.safe_input', return_value='1'):
+            result = main.main_menu()
+        self.assertEqual(result, main.add_new_item, "Invalid main menu result for option 1")
+
+    @mock.patch('main.market_prices')
+    @mock.patch('main.management')
+    def test_add_new_item_inventory(self, mock_mangement, mock_market_prices):
+        main.add_new_item(properties=self.properties)
+        mock_mangement.add_inventory.assert_called_with(self.properties, mock_market_prices)
+
+    @mock.patch('main.market_prices')
+    @mock.patch('main.management')
+    def test_add_new_item_furniture(self, mock_mangement, mock_market_prices):
+        self.properties["is_furniture"] = "y"
+        main.add_new_item(properties=self.properties)
+        mock_mangement.add_furniture.assert_called_with(self.properties, mock_market_prices)
+
+    @mock.patch('main.market_prices')
+    @mock.patch('main.management')
+    def test_add_new_item_electric_appliance(self, mock_mangement, mock_market_prices):
+        self.properties["is_electric_appliance"] = "y"
+        main.add_new_item(properties=self.properties)
+        mock_mangement.add_electric_appliance.assert_called_with(self.properties, mock_market_prices)
+
+
+    def test_safe_input(self):
+        with mock.patch('builtins.input', return_value='test_return_value') as _raw_input:
+            self.assertEqual(main.safe_input("Enter a value: "), "test_return_value")
+            _raw_input.assert_called_once_with('Enter a value: ')
+            
+
+    @mock.patch('main.management')
+    def test_item_info_not_found(self, mock_management):
+        mock_management.get_item = MagicMock(return_value=None)
+        with captured_output() as (out, err):
+            main.item_info("test")
+        result = out.getvalue().strip()
+        self.assertEqual(result, "Item not found in inventory")
+
+    @mock.patch('main.management')
+    def test_item_info_returned_value(self, mock_management):
+        mock_management.get_item = MagicMock(return_value={"test_key": "test_val"})
+        with captured_output() as (out, err):
+            main.item_info("test")
+        result = out.getvalue().strip()
+        self.assertEqual(result, "test_key:test_val")
+    
+    @mock.patch('main.management')
+    def test_item_info_no_input(self, mock_management):
+        mock_management.get_item = MagicMock(return_value={"test_key": "test_val"})
+        with mock.patch('main.safe_input', return_value='my_test_value'):
+            with captured_output() as (out, err):
+                main.item_info()
+            result = out.getvalue().strip()
+        self.assertEqual(result, "test_key:test_val")
```

Comment:
Great job with these tests. I think you understand mock pretty well now. :) User interfaces are a pain in the butt to test.